### ISSUE TITLE
v2.5.0 throws DeadlineExceeded, pinning action back to v2.4.1 works fine

### ISSUE THREAD

**Issue by fredvd** [docker/setup-buildx-action#222]
Three days ago our nightly image generation workflow on  started failing.  https://github.com/plone/plone-frontend/actions/runs/4405764859

I tried to restart it today a few times, consistent failure. When I pin setup-buildx-action from v2  in our workflow file back to exactly v2.4.1 the build step works again. Full error:

```
#3 resolve image config for docker.io/docker/dockerfile:1
#3 ERROR: failed to do request: Head "https://registry-1.docker.io/v2/docker/dockerfile/manifests/1": dial tcp 34.205.13.154:443: i/o timeout
------
 > resolve image config for docker.io/docker/dockerfile:1:
------
Dockerfile.nightly:1
--------------------
   1 | >>> # syntax=docker/dockerfile:1
   2 |     FROM node:16-slim as base
   3 |     FROM base as builder
--------------------
ERROR: failed to solve: DeadlineExceeded: DeadlineExceeded: DeadlineExceeded: failed to do request: Head "https://registry-1.docker.io/v2/docker/dockerfile/manifests/1": dial tcp 34.205.13.154:443: i/o timeout
Error: buildx failed with: ERROR: failed to solve: DeadlineExceeded: DeadlineExceeded: DeadlineExceeded: failed to do request: Head "https://registry-1.docker.io/v2/docker/dockerfile/manifests/1": dial tcp 34.205.13.154:443: i/o timeout
```

Local Docker/setup info from setup-buildx, self hosted arm64 runner

```
 /usr/local/bin/docker version
  Client:
   Cloud integration: v1.0.31
   Version:           20.10.23
   API version:       1.41
   Go version:        go1.18.10
   Git commit:        7155243
   Built:             Thu Jan 19 17:35:19 2023
   OS/Arch:           darwin/arm64
   Context:           default
   Experimental:      true
  
  Server: Docker Desktop 4.17.0 (99724)
   Engine:
    Version:          22.06.0-beta.0-926-g914b02ebaf.m
    API version:      1.43 (minimum version 1.12)
    Go version:       go1.18.4
    Git commit:       914b02ebaf
    Built:            Thu Feb  9 12:31:06 2023
    OS/Arch:          linux/arm64
    Experimental:     false
   containerd:
    Version:          1.6.18
    GitCommit:        2456e983eb9e37e47538f59ea18f2043c9a73640
   runc:
    Version:          1.1.4
    GitCommit:        v1.1.4-0-g5fd4c4d
   docker-init:
    Version:          0.19.0
    GitCommit:        de40ad0
  /usr/local/bin/docker info
  Client:
   Context:    default
   Debug Mode: false
   Plugins:
    buildx: Docker Buildx (Docker Inc., v0.10.3)
    compose: Docker Compose (Docker Inc., v2.15.1)
    dev: Docker Dev Environments (Docker Inc., v0.1.0)
    extension: Manages Docker extensions (Docker Inc., v0.2.18)
    sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
    scan: Docker Scan (Docker Inc., v0.25.0)
    scout: Command line tool for Docker Scout (Docker Inc., v0.6.0)
  
  Server:
   Containers: 1
    Running: 1
    Paused: 0
    Stopped: 0
   Images: 2
   Server Version: 22.06.0-beta.0-926-g914b02ebaf.m
   Storage Driver: stargz
    driver-type: io.containerd.snapshotter.v1
   Logging Driver: json-file
   Cgroup Driver: cgroupfs
   Cgroup Version: 2
   Plugins:
    Volume: local
    Network: bridge host ipvlan macvlan null overlay
    Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
   Swarm: inactive
   Runtimes: io.containerd.runc.v2 runc
   Default Runtime: runc
   Init Binary: docker-init
   containerd version: 2456e983eb9e37e47538f59ea18f2043c9a73640
   runc version: v1.1.4-0-g5fd4c4d
   init version: de40ad0
   Security Options:
    seccomp
     Profile: builtin
    cgroupns
   Kernel Version: 5.15.49-linuxkit
   Operating System: Docker Desktop
   OSType: linux
   Architecture: aarch64
   CPUs: 4
   Total Memory: 7.668GiB
   Name: docker-desktop
   ID: 2a44d730-7c94-411c-851b-205580978fb2
   Docker Root Dir: /var/lib/docker
   Debug Mode: false
   HTTP Proxy: http.docker.internal:3128
   HTTPS Proxy: http.docker.internal:3128
   No Proxy: hubproxy.docker.internal
   Registry: https://index.docker.io/v1/
   Labels:
   Experimental: false
   Insecure Registries:
    hubproxy.docker.internal:5000
    127.0.0.0/8
   Live Restore Enabled: false
```

**Comment by crazy-max** [comment]
Doesn't look linked to the action but your self-hosted environment. I see that before your server was on `Docker Engine 20.10.23`: https://github.com/plone/plone-frontend/actions/runs/4383069741/jobs/7672865661#step:5:37. But now you are on an unstable `Docker Engine 22.06.0-beta.0-926-g914b02ebaf.m` with [`io.containerd.snapshotter.v1` enabled](https://github.com/plone/plone-frontend/actions/runs/4405764859/jobs/7717080961#step:5:75) which is not yet released and supported yet. Suggest to review your environment.

**Comment by fredvd** [comment]
Doesn't look like you checked more than one run. 

The switch to unstable was a test I did today in the Docker desktop setting to use experimental/new features.  The failed nightlies from the previous days all use stable 20.20.23. , with the new v2.5.0 action.

#327 https://github.com/plone/plone-frontend/actions/runs/4399721843/jobs/7716052837
Version:          22.06.0-beta.0-926-g914b02ebaf.m
5 hours ago

#326 https://github.com/plone/plone-frontend/actions/runs/4394442334/jobs/7695495273
DVersion:          20.10.23
yesterday

#325 https://github.com/plone/plone-frontend/actions/runs/4388927174
Version:          20.10.23
2 days ago

**Comment by crazy-max** [comment]
> The failed nightlies from the previous days all use stable 20.20.23. , with the new v2.5.0 action.

I'm still not sure how this is related to the action. There is no functional change you're using that could affect your build.

* https://github.com/plone/plone-frontend/actions/runs/4369062434/jobs/7642421365 from 5 days ago for example has the same error and was using the 2.4.1 version.
* Same for https://github.com/plone/plone-frontend/actions/runs/4359097037/jobs/7620491106#step:1:27 7 days ago.

After discussing internally this seems to be an issue with Docker Desktop that will be fixed on next release.

**Comment by fredvd** [comment]
> I'm still not sure how this is related to the action. There is no functional change you're using that could affect your build.

I'm also not sure myself, we're still relatively new to using self hosted runners, but kind of forced because of the 3+ hours of runtime on the github x86 runners when building arm64 images.  

The only thing I saw that changed 2 days ago was the action version so I tried pinning that down.  But this evening we had another failure.  When I checked the runner desktop I noticed that the docker hub user I created for this runner was logged out.  I first thought that re-using the same docker hub account on mutiple machines might be a cause, but that doesn't seem to be the case. 

I restarted our failed job and I observed the runner Mac desktop this time: I see now that my personal user account gets logged out from docker desktop, and it logs in with the service user under which the setupx-build action will eventually push the finished action to docker hub (dockerplone).  30-60 seconds after that login the workflow step fails again with thie DeadlineExceeded message. 

If there is anything else I can test or provide to help with this issue, please let me know.

**Comment by fredvd** [comment]
> After discussing internally this seems to be an issue with Docker Desktop that will be fixed on next release.

Are there any work arounds or advice for this issue with Docker desktop itself? I had to restart our worker twice now, because all workflows using the buildx action start to fail.  Somtimes a restart helps, sometimes it doesn't and I have no clue what really helps. 

Also, can we keep this issue open until it is resolved so other setup-buildx-action users can see this while it is relevant? I was a bit surprised and disappointed to find this issue that I reported closed within a few hours that I created it, before there was any discussion on what is going on.
